lib/rules/no-cond-assign.js

```diff
@@ -59,7 +59,7 @@ module.exports = {
                 if (isConditionalTestExpression(currentAncestor)) {
                     return currentAncestor.parent;
                 }
-            } while ((currentAncestor = currentAncestor.parent));
+            } while ((currentAncestor = currentAncestor.parent) && currentAncestor.type !== "CallExpression");
```

Comment:
Good call. I'll fix it. 

> On Aug 15, 2016, at 1:25 PM, Kevin Partington <notifications@github.com> wrote:
> 
> In lib/rules/no-cond-assign.js:
> 
> > @@ -59,7 +59,7 @@ module.exports = {
> >                  if (isConditionalTestExpression(currentAncestor)) {
> >                      return currentAncestor.parent;
> >                  }
> > -            } while ((currentAncestor = currentAncestor.parent));
> > +            } while ((currentAncestor = currentAncestor.parent) && currentAncestor.type !== "CallExpression");
> Shouldn't this say "FunctionExpression" instead, in that case? "CallExpression" is for actual function calls.
> 
> In the case of anArray.some(function () { a = 5; }), the FunctionExpression is just the argument, but the CallExpression is the whole example.
> 
> I don't think we should address the CallExpression problem (doSomething(a = 5)) in this issue/PR. The original issue/PR was about an assignment happening in a different scope. The scope is introduced by the FunctionExpression, not the CallExpression.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>